feature-libs/checkout/base/core/facade/checkout-payment.service.spec.ts

```diff
@@ -0,0 +1,338 @@
+  }
+}
+
+describe(`CheckoutPaymentService`, () => {
+  let service: CheckoutPaymentService;
+  let connector: CheckoutPaymentConnector;
+  let store: MockStore;
+  let checkoutQuery: CheckoutQueryFacade;
+  let eventService: EventService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        CheckoutPaymentService,
+        provideMockStore(),
+        { provide: ActiveCartService, useClass: MockActiveCartService },
+        { provide: UserIdService, useClass: MockUserIdService },
+        { provide: EventService, useClass: MockEventService },
+        {
+          provide: CheckoutPaymentConnector,
+          useClass: MockCheckoutPaymentConnector,
+        },
+        { provide: CheckoutQueryFacade, useClass: MockCheckoutQueryFacade },
+      ],
+    });
+
+    service = TestBed.inject(CheckoutPaymentService);
+    connector = TestBed.inject(CheckoutPaymentConnector);
+    store = TestBed.inject(MockStore);
+    checkoutQuery = TestBed.inject(CheckoutQueryFacade);
+    eventService = TestBed.inject(EventService);
+  });
+
+  it(`should inject CheckoutPaymentService`, inject(
+    [CheckoutPaymentService],
+    (checkoutPaymentService: CheckoutPaymentService) => {
+      expect(checkoutPaymentService).toBeTruthy();
+    }
+  ));
+
+  describe(`getCardTypesState`, () => {
+    it(`should call the checkoutPaymentConnector.getCardTypes()`, (done) => {
+      spyOn(connector, 'getCardTypes').and.callThrough();
+
+      service
+        .getCardTypesState()
+        .pipe(take(1))
+        .subscribe((state) => {
+          expect(state).toEqual({
+            loading: false,
+            error: false,
+            data: mockCardTypes,
+          });
+          expect(connector.getCardTypes).toHaveBeenCalled();
+          done();
+        });
+    });
+
+    it(`should unsuccessfully backOff on Jalo error and put the error in the state`, fakeAsync(() => {
+      spyOn(connector, 'getCardTypes').and.returnValue(
+        throwError(mockJaloError)
+      );
+
+      let resultState: QueryState<CardType[] | undefined> | undefined;
+      const subscription = service
+        .getCardTypesState()
+        .subscribe((result) => (resultState = result));
+
+      tick(4200);
+
+      expect(resultState).toEqual({
+        loading: false,
+        error: <Error>mockJaloError,
+        data: undefined,
+      });
+      subscription.unsubscribe();
+    }));
+
+    xit(`should successfully backOff on Jalo error and recover after the 2nd attempt`, fakeAsync(() => {
+      spyOn(connector, 'getCardTypes').and.returnValues(
+        // first attempt
+        throwError(mockJaloError),
+        // second attempt
+        throwError(mockJaloError),
+        // third time the charm
+        of(mockCardTypes)
+      );
+
+      let resultState: QueryState<CardType[] | undefined> | undefined;
+      const subscription = service.getCardTypesState().subscribe((result) => {
+        console.log('result', result);
```

Comment:
reminder for me

Yeah, kind of weird. Tried to fix/figure it out, but I'll check it tomorrow again

Reply:
Removed the console log.
And this is something that I wanted to talk to you, there's one issue here. I have a todo on my list, and we can discuss it in a separate PR.